hydra-cluster/test/Test/GeneratorSpec.hs

```diff
@@ -27,19 +33,25 @@ prop_computeValueFromUTxO =
 
 prop_keepsUTxOConstant :: Property
 prop_keepsUTxOConstant =
-  forAll arbitrary $ \(Positive n) ->
-    forAll (genConstantUTxODataset defaultProtocolParameters n) $ \Dataset{fundingTransaction, transactionsSequence} ->
-      let initialUTxO = utxoFromTx fundingTransaction
-          finalUTxO = foldl' apply initialUTxO transactionsSequence
-       in length finalUTxO == length initialUTxO
-            & counterexample ("\ntransactions: " <> jsonString transactionsSequence)
-            & counterexample ("\nutxo: " <> jsonString initialUTxO)
+  forAll arbitrary $ \(Positive n) -> do
+    idempotentIOProperty $ do
```

Comment:
discovery: Did not know about this function, looks interesting and useful.

Reply:
Me neither.. before yesterday :)